### DISCUSSION TITLE
Are subpages links always broken?

### DISCUSSION THREAD

**Discussion by Mapaor4** [transitive-bullshit/nextjs-notion-starter-kit#613] [Q&A]
I know this repo is designed to make a blog. But what if instead I want to build a knowledge base website (something between a wiki and a documentation page)? 

If I create a subpage inside Notion, it deploys just fine in Vercel (the icon and the name of the subpage are there) but when I click it I get 404 error (notion page not found).

I understand this is because that page is not on the database that the API fetches content from. But is there a way to include subpages that are not blog entries? (even if it's through manually changing each link to a hidden blog entry)

Thank you

**Comment by eddiehe99** [comment]
Actually, you may build any website using this project. This project is based on react-notion-x which can transform your Notion page into a website just like what the Super.so does.

Subpages is supported in this project. For instance, the [About](https://nextjs-notion-starter-kit.transitivebullsh.it/about) page (roll down to the bottom on the [demo Notion page](https://transitive-bs.notion.site/Next-js-Notion-Starter-Kit-Template-7875426197cf461698809def95960ebf) and you will find this page) in the template Notion page is a subpage. If you get any error (like 404), try to redeploy your project.

**Reply by kufx** [reply]
Actually, it only requires modifying the link format of the page. The default page link suffix is the page title. Simply changing the link suffix to the Notion page ID will solve the problem!I am asking about the solutions obtained by AI. For specific details, please refer to my article: https://notes.kuhehe.top/1ada7a4ad19180edb371e9fec78d0b1d#33aa7a4ad19180c28394cf88cc4aa1a8

**Comment by elenimou** [comment]
I also had this issue where subpages (grandchildren of the main page) worked in dev mode but 404'd in prod. 

I was able to fix this by updating the `maxDepth` parameter in `get-site-map.ts`.

```
async function getAllPagesImpl(
  rootNotionPageId: string,
  rootNotionSpaceId?: string,
  {
    maxDepth = 3
  }: {
    maxDepth?: number
  } = {}
): Promise<Partial<types.SiteMap>> {
  const pageMap = await getAllPagesInSpace(
    rootNotionPageId,
    rootNotionSpaceId,
    getPage,
    {
      maxDepth
    }
  )
  ```
  
  ^^^ The original maxDepth was 1, but when I changed it to 3 like above it fixed the issue for me.

**Reply by kufx** [reply]
Actually, it only requires modifying the link format of the page. The default page link suffix is the page title. Simply changing the link suffix to the Notion page ID will solve the problem!I am asking about the solutions obtained by AI. For specific details, please refer to my article: https://notes.kuhehe.top/1ada7a4ad19180edb371e9fec78d0b1d#33aa7a4ad19180c28394cf88cc4aa1a8

**Comment by kufx** [comment]
Actually, it only requires modifying the link format of the page. The default page link suffix is the page title. Simply changing the link suffix to the Notion page ID will solve the problem!I am asking about the solutions obtained by AI. For specific details, please refer to my article: https://notes.kuhehe.top/1ada7a4ad19180edb371e9fec78d0b1d#33aa7a4ad19180c28394cf88cc4aa1a8
